test.py

```diff
@@ -365,13 +369,19 @@ async def create_cluster():
         else:
             raise RuntimeError("Unsupported topology name")
 
-    async def add_test(self, shortname) -> None:
-        test = PythonTest(self.next_id, shortname, self)
-        self.tests.append(test)
+    def build_test_list(self) -> List[str]:
+        """For pytest, search for directories recursively"""
+        path = self.suite_path
+        pytests = itertools.chain(path.rglob("*_test.py"), path.rglob("test_*.py"))
+        return [os.path.splitext(t.relative_to(self.suite_path))[0] for t in pytests]
 
     @property
     def pattern(self) -> str:
-        return "test_*.py"
+        assert False
```

Comment:
It's a bit strange that you have this "pattern" thing for tests, but in the pytest tests it doesn't get called, and instead "build_test_list()" get called - it suggests the design of this of this class is wrong. But it can be cleaned up later.